Removed restriction to allow clusters of a single node.

Stexxen opened the pull request:
Default functionality remains the same, but restriction can be lifted with a config option.
Also created Example to demonstrate new option and clusters of clusters.

Stexxen commented:
@mojoaxel thanks for merging my other pull.
This one is not a bug fix but a new feature, or rather the removal of a restriction. I've created an example, but don't think this pull entirely complies with misc/how_to_publish.md. Do you have any advice on what else I need to do?
Thanks,
S.

mojoaxel commented on examples/network/other/clustersOfclusters.html:
```diff
@@ -0,0 +1,77 @@
+            }
+        });
+        cluster();
+        cluster();
```
One of this calls should be enough to show the new feature ;-)

mojoaxel commented:
@Stexxen Very nice! Why not!
Can you please add docs for the new option `clusterNodeProperties` [here](/almende/vis/blob/develop/docs/network/index.html#L1186). 
You just can commit changes in your `single_cluster` branch to update this pull-request.

After that I see no reason not to merge this.

mojoaxel commented:
@Stexxen I still think the new option `allowSingleNodeCluster` should be documented!?

mojoaxel commented:
@Stexxen Can you please add some documentation. If you can not do it please let us know so we can do it. Thanks!

Stexxen commented:
apologies, I will work on this today.

Stexxen commented:
@mojoaxel documentation now added, and removed a couple of the calls to cluster(). I've still left 3 in as the example is called clusterOfClusters after all ;-)

mojoaxel commented on docs/network/index.html:
```diff
@@ -1181,7 +1181,18 @@ <h4 id="optionsObject">Cluster methods options object</h4>
+                    fine tuning. If undefined, default node options will be used.<br/><br/>
+                    Default functionality only allows clustering if the cluster contains 2 or more nodes. To allow clustering of single nodes you can use the <code>allowSingleNodeCluster : true</code> property.
+<pre class="prettyprint lang-js">
+var clusterOptions = {
```
you should call it "options" to be coherent with the other examples

Stexxen replied:
ok

mojoaxel commented on docs/network/index.html:
```diff
@@ -1181,7 +1181,18 @@ <h4 id="optionsObject">Cluster methods options object</h4>
+                    Default functionality only allows clustering if the cluster contains 2 or more nodes. To allow clustering of single nodes you can use the <code>allowSingleNodeCluster : true</code> property.
+<pre class="prettyprint lang-js">
+var clusterOptions = {
+    joinCondition: function (potentialNodeOptions) {
```
The "joinCondition" is from a other example ans should be removed.

Stexxen replied:
"joinCondition" is not always optional. Thats why I chose to leave it in, so it is a complete example, but please confirm you want it removed?
